src/placeos-core/module_manager.cr

```diff
@@ -151,6 +153,19 @@ module PlaceOS
       Log.info { {message: "started module", module_id: mod.id, name: mod.name, custom_name: mod.custom_name} }
     end
 
+    def restart_module(mod : Model::Module)
+      mod_id = mod.id.as(String)
+      manager = proc_manager_by_module?(mod_id)
+
+      if manager
+        manager.stop(mod_id)
+        start_module(mod)
+        Log.info { {message: "restarted module", module_id: mod.id, name: mod.name, custom_name: mod.custom_name} }
+      else
+        Log.error { {message: "missing protocol manager on restart", module_id: mod.id, name: mod.name, custom_name: mod.custom_name} }
+      end
+    end
```

Comment:
Any reason this can't just call `stop_module` then `start_module`?

If it's due to logging only, currently this looks like it will stop the module silently, then start it again (printing normal start logging), following by the "restarted module" entry. To make this a little easier to follow it's preferable to either print print all events, or just the single restarted summary.

Reply:
`stop_module` will remove the driver binary if there's no more references to it.

Reply:
could log that we're restarting before calling `manager.stop(mod_id)`